File: Phase Gate Project/credit_card.py
def get_modified_string(card):
	modified_card_number = card.replace(" ", "").replace("_", "")
	length = len(modified_card_number)

	if 13 <= length <= 16:
		if modified_card_number.startswith("4"):
			card_type = "Visa Card"
			print("**Credit Card Number: "+str(length))
		elif modified_card_number.startswith("5"):
			card_type = "MasterCard"
			print("**Credit Card Number: "+str(length))
		elif modified_card_number.startswith("37"):
			card_type = "American Express Card"
			print("**Credit Card Number: "+str(length))
		elif modified_card_number.startswith("6"):
			card_type = "Discover Card"
			print("**Credit Card Number: "+str(length))
		else:
			card_type = "Invalid Card"
	else:
		return "Invalid Credit Card!"

	return modified_card_number, card_type, length


def get_converted_int_credit_card(modified_card):
	return int(modified_card)


def get_odd_and_even_indexes(modified_card):
	data = str(get_converted_int_credit_card(modified_card))
	array_double_converted_int_credit_card = []

	for index, digit in enumerate(data):
		num = int(digit)
		if (len(data) % 2 == 0 and index % 2 == 0) or (len(data) % 2 == 1 and index % 2 != 0):
			double_num = num * 2
			if double_num > 9:
				double_num -= 9
				array_double_converted_int_credit_card.append(double_num)
			else:
				array_double_converted_int_credit_card.append(num)

	return array_double_converted_int_credit_card


def check_validity(modified_card):
	check_validity = get_odd_and_even_indexes(modified_card)
	total_array_sum = sum(check_validity)

	if total_array_sum % 10 == 0:
		return "**Credit Card Validity Status: Valid"
	else:
		return "**Credit Card Validity Status: Invalid"


card_number = input("Hello, kindly enter card details to verify: ")
modified_card, card_type, length = get_modified_string(card_number)

if modified_card:
	print(f"**Credit Card Type: {card_type}")
	print(f"**Credit Card Digit Length: {length}")
	validity_status = check_validity(modified_card)
	print(validity_status)

File: Phase Gate Project/CreditCard.java
import java.util.Scanner;

public class CreditCard{

	public static void main(String[] args) {

		Scanner scanner = new Scanner(System.in);
		
		try {

			System.out.println("Hello, Kindly Enter Card details to verify");
			String cardNumber = scanner.nextLine();

			System.out.print(checkValidity(cardNumber));
		}
		
		catch (Exception e) {
		
			System.out.println("Invalid Input");

		}

	}
		
	public static String getModifiedString(String card){

		String aboutToModify = card;

		String modifiedCardNumber = aboutToModify.replace(" ", "").replace("_", "");

		int length = modifiedCardNumber.length();
		
		if(length >= 13 && length <= 16){

			if(modifiedCardNumber.startsWith("4")) {
	
				System.out.println("**Credit Card Type: Visa Card");

				System.out.println("**Credit Card Digit Length "+length);

				System.out.println("**CreditCard Number: "+ modifiedCardNumber);

			} 

			else if (modifiedCardNumber.startsWith("5")) {

				System.out.println( "**Credit Card Type: MasterCard Card");

				System.out.println("**Credit Card Digit Length "+length);

				System.out.println("**CreditCard Number: "+ modifiedCardNumber);

			} 
			else if (modifiedCardNumber.startsWith("37")) {

				System.out.println("**Credit Card Type: American Express Card");

					System.out.println("**Credit Card Digit Length "+length);

					System.out.println("**CreditCard Number: "+ modifiedCardNumber);

			} 
			else if (modifiedCardNumber.startsWith("6")) {

				System.out.println("**Credit Card Type: Discover Card");

					System.out.println("**Credit Card Digit Length "+length);

					System.out.println("**CreditCard Number: "+ modifiedCardNumber);

				}
			else {

				System.out.println("**Credit Card Type: Invalid Card");

				System.out.println("**Credit Card Digit Length "+length);

				System.out.println("**CreditCard Number: "+ modifiedCardNumber);

			}

		}

		else{
			System.out.println("Invalid Credit Card!");

		}

		return modifiedCardNumber;

	}	
	
	
	public static int[] getConvertedIntCreditCard(String modifiedCard){

		String alreadyModified = getModifiedString(modifiedCard);

		int[] convertedIntCreditCard = new int[alreadyModified.length()];
			
		for(int count = 0; count < alreadyModified.length(); count++){

			convertedIntCreditCard[count] = Integer.valueOf(String.valueOf(modifiedCard.charAt(count)));

		} 
			
		return convertedIntCreditCard;

	}


	public static int[] getOddAndEvenIndexes(String elements){

		int[] data = getConvertedIntCreditCard(elements);

		int[] arrayDoubleConvertedIntCreditCard = new int[data.length];

		int doubleConvertedIntCreditCard = 0;

		for(int count = 0; count < data.length; count++){

			if(data.length % 2 == 0){

				if(count % 2 == 0){

					doubleConvertedIntCreditCard = data[count] * 2;

					if(doubleConvertedIntCreditCard > 9){
									
						doubleConvertedIntCreditCard -= 9;		

					}
						
					else{

						doubleConvertedIntCreditCard = data[count] * 2;

					}

					arrayDoubleConvertedIntCreditCard[count] = doubleConvertedIntCreditCard;
							
				}

				else{

					arrayDoubleConvertedIntCreditCard[count] = data[count];

				}
			}


			else if (data.length % 2 == 1){

				if(count % 2 != 0){

					doubleConvertedIntCreditCard = data[count] * 2;

					if(doubleConvertedIntCreditCard > 9){
									
						doubleConvertedIntCreditCard -= 9;					
		
					}

					else{

						doubleConvertedIntCreditCard = data[count] * 2;

					}

					arrayDoubleConvertedIntCreditCard[count] = doubleConvertedIntCreditCard;
				}
			}

			else{

				arrayDoubleConvertedIntCreditCard[count] = data[count];
			}
								
		}

				
		return arrayDoubleConvertedIntCreditCard;

	}
	

	public static String checkValidity(String validity){

		int[] checkValidity = getOddAndEvenIndexes(validity);

		int totalArraySum = 0;

		String result = "";

		for(int count = 0; count < checkValidity.length; count++){

			totalArraySum += checkValidity[count];
			
		}

		if(totalArraySum % 10 == 0){
				
			result = "**Credit Card Vadility Status: Valid";

		}

		else{

			result = "**Credit Card Vadility Status: Invalid";

		}

		return result;
	}

		

}